src/CMakeLists.txt

```diff
@@ -758,17 +758,19 @@ else()
   endif()
 endif()
 
-# Get the LLVM libraries we need
-llvm_map_components_to_libnames(LIBS ${LLVM_COMPONENTS})
-
-# When building a shared library the LLVM libraries will be
-# embedded in the Halide library. When building a static library
-# LLVM is not embedded but CMake knows that when building an executable
-# against the Halide static library that it needs to link LLVM too so
-# PRIVATE scope is the correct choice here.
-target_link_libraries(Halide PRIVATE ${LIBS})
+# When building a shared library the LLVM libraries will either be
+# embedded or linked into the Halide library. When building a static library
+# LLVM is not embedded/linked but CMake knows that when building an executable
+# against the Halide static library that it needs to link LLVM too, IFF the
+# halide is being built as a subproject. (which means, statically-built halide
```

Comment:
Nit: make a proper sentence: "Which means, statically-built Halide doesn't work out-of-installation directory."